### DISCUSSION TITLE
Plotly plots are not rendering

### DISCUSSION THREAD

**Discussion by charldelaoa** [quarto-dev/quarto-cli#5264] [Q&A] [answered]
Hi, I am trying to render plotly plots using python + quarto. 

I keep getting this message:

Unable to display output for mime type(s): application/vnd.plotly.v1+json

Is there something I need to modify/add to the yml header?

![Screenshot 2023-04-21 at 17 28 22](https://user-images.githubusercontent.com/120415887/233687693-a21844b5-a9c7-4c52-ad4a-e8ca6a9fabea.png)

**Comment by mcanouil** [comment]
Could you share a complete and small reproducible Quarto document?
Could you also state which version of Quarto and which operating system you are using?

**Reply by charldelaoa** [reply]
Hi @mcanouil, 

I'm actually trying to render a Jupyter notebook .ipynb. 
 I have been successful rendering plots using other libraries (e.g., Bokeh) and was able to create a book from several .ipynb files using quarto (it's amazing)!:
https://charldelao.quarto.pub

I am able to render the plotly plot in the notebook but it does not generate the html plot.
- I am running quarto 1.3.336 in MacIC Ventura 13.1


[Plotlyquarto.zip](https://github.com/quarto-dev/quarto-cli/files/11308491/Plotlyquarto.zip)

**Reply by mcanouil** [reply]
Do you have the example as a Git repository rather than a zip?

**Reply by charldelaoa** [reply]
https://github.com/charldelaoa/quarto-plotly

**Reply by charldelaoa** [reply]
@mcanouil I made a .qmd file instead of a .ipynb with the same code and I was able to render the plotly plots. Maybe there is something going wrong when trying to render directly from Jupyter notebooks?

**Reply by cscheid** [reply, marked as answer]
If you're calling `quarto render file.ipnyb` you need to tell quarto explicitly in case you want to execute the notebook with `--execute`. This is explained [in our documentation](https://quarto.org/docs/projects/code-execution.html#notebooks)

**Reply by prem308** [reply]
Thank you so much

**Comment by cderv** [comment]
@charldelaoa one of our  "getting started" with python is using  a plotly graph. https://quarto.org/docs/get-started/computations/text-editor.html

I tested this document and it works ok.  As @mcanouil we need an example to see exactly the python code you used. 

Also one precision based on guess from your screenshot: 
* Is this happening in VSCODE inline preview ? 
* Is this happening with `quarto render` output too ? 

I am asking because it seems in the former case, this is something that could be related to VSCODE extension 
* https://github.com/microsoft/vscode-jupyter/issues/11658
* https://github.com/microsoft/vscode-jupyter/issues/4227
* https://stackoverflow.com/questions/74046983/plotly-no-renderer-could-be-found-for-mimetype-application-vnd-plotly-v1json

So wondering if this is happening with Quarto itself, or more something related to VSCODE / Quarto extension, and other integration

**Reply by charldelaoa** [reply]
Hi @cderv,

I am able to see VSCODE inline preview using .qmd files

The issue was when trying to render .ipynb to .html as the plotly plots would not render.

I have created a .qmd file instead and that seems to have fix the issue

**Comment by ForBo7** [comment]
Running the following line in your notebook should fix it.
```py
import plotly.io as pio
pio.renderers.default = "notebook"
```

**Comment by mladencucakSYN** [comment]
I am facing the same issue here, rendering quarto with R and several plotly plots. Some of them render and some do not.

**Reply by mcanouil** [reply]
Did you try what was suggested above?

If yes, without any information except "I have the same problem", I am afraid there is nothing that can be done to help.

Also, as you said "R", you're issue might have nothing to do with Quarto but instead related to `reticulate`. You could check for that, by trying to render using `rmarkdown` instead of Quarto.

Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by mladencucakSYN** [reply]
Unfortunately data is confidential. 
This is my header 
```
---
title: "Data"
format: 
  html:
    # code-fold: show
    out.width: "100%"
editor: visual
# runtime: shiny
execute:
  cache: true
  echo: false
---

``` 
Imagine plotting mtcars for each cylinder and first plot works, the second doesn't, the third does and the fourth does. These are plots made with ggplot. all works in Rmd

**Reply by mcanouil** [reply]
I did not ask to share your data but to make a reproducible example.
Unfortunately, there is nothing we can do with what you shared.

Please once you have something we can work on, do open another discussion as your case is not related to this thread which uses Jupyter, not knitr/R.

**Comment by benoit-cty** [comment]
Hello,
For the record, it is because VSCode (at least with [Jupyter Notebook Renderers](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter-renderers)) do not render Plotly in the same way than Jupyter did.

In a file saved by Jupyter the cell output is Javascript content with the code of plotly.

In a file saved by VSCode, there is only the data needed by plotly, not plotly itself:
```
{
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.plotly.v1+json": {
       "config": {
        "plotlyServerURL": "https://plot.ly"
       },
       "data": [...
```

It save a lot of space but it is not supported by Quarto.

Adding 
```
import plotly.io as pio
pio.renderers.default = "notebook"
```
In VSCode force it to use the same output as Jupyter so it works with Quarto.

**Comment by Hamsterarsch** [comment]
For anyone arriving here recently and for whom the [notebook mentioned above](https://quarto.org/docs/get-started/computations/text-editor.html) does not produce plotly output: There was a breaking change in plotly-6.0 (see #12061 ).

downgrade plotly using `pip install --force-reinstall -v="plotly-5.24"`

**Reply by mcanouil** [reply]
FYI, the team is aware:

- https://github.com/quarto-dev/quarto-cli/issues/12061

**Comment by LCPallares** [comment]
i used:
```python
import plotly.offline as pyo
pyo.init_notebook_mode(connected=True)
```
